lib/src/kbasesearchengine/main/SearchMethods.java

```diff
@@ -295,6 +295,37 @@ public SearchObjectsOutput searchObjects(SearchObjectsInput params, String user)
         ret.withSearchTime(System.currentTimeMillis() - t1);
         return ret;
     }
+     //for testing purposes. Access set to admin
```

Comment:
C&Ping this method to merely add admin creds seems like a bad idea. Firstly, if we really need an admin method, the right thing to do is modify the source method to take an asAdmin boolean or something like that and set up the access filter based on said boolean. However, I don't really understand why that's needed at all - is it not possible to call the method with proper input so the user gets treated as an admin?

Further though - all the changes are in the ElasticIndexingStorage class so why not just write tests against that class directly instead of indirectly via SearchMethods?

Reply:
I actually started writing it in elasticindexingstorage but I needed the same classes but didn't need to invoke all the parameters. so I ended up moving things out and making mocks. And since I was doing that already, it was easier to call on SearchMethods directly so that I didn't have to copy most of the SearchObjectsOutput method in the setup. And yeah I ended up having to copy the method b/c of the accessFilter is kind of messed up. It takes a username and changes it from kbaseengine.AccessFilter to kbaseengine.search.AccessFilter <= this one can be set as admin. I could move it back in?

Reply:
I don't think we should be copying searchObjects anywhere - we should be doing unit tests against EIS similar to the other unit tests.